### PR TITLE
add reader_params

### LINKED ISSUE #214: [Manifest] Add reader_params to add_url
add reader_params as a kwarg to add_url() so that specific reader configuration parameters can be passed

https://github.com/solvebio/solvebio-python/blob/master/solvebio/resource/manifest.py#L43

### PR DESCRIPTION
**Pull request by jsh2134**
fixes #214

### CODE REVIEW

**Comment by jsh2134** [comment]
actually will not support this right now since the `reader_params` can just be sent with the DatasetImport

**Comment by jsh2134** [comment]
actually this may not be the case. this needs to be confirmed.

**Comment by jsh2134** [comment]
ready for review. API now correctly supports all the manifest parameters, supports pulling them from DatasetImport if not defined in the manifest AND supports overriding them per file in the manifest _should_ someone have a case where they want to do that.

File: solvebio/resource/manifest.py [hunk 1]
```diff
@@ -37,18 +37,16 @@ def add_file(self, path):
...
     def add_url(self, url, **kwargs):
-        self.manifest['files'].append({
-            'url': url,
-            'name': kwargs.get('name'),
-            'format': kwargs.get('format'),
-            'size': kwargs.get('size'),
-            'md5': kwargs.get('md5'),
-            'base64_md5': kwargs.get('base64_md5')
-        })
+        manifest_item = dict(url=url)
```
Comment (davecap) [inline solvebio/resource/manifest.py]: Minor point but you `dict(url=url, **kwargs)` should work

**Pull request merged**